### DISCUSSION TITLE
gitea support for integrating a comment system

### DISCUSSION THREAD

**Discussion by NejatHakan** [squidfunk/mkdocs-material#7842] [Q&A]
Hi,

in the official doc --> https://squidfunk.github.io/mkdocs-material/setup/adding-a-comment-system/
we can find a verry nice documentation on how to implement a comment system with giscus and github into mkdocs material.

But I would rather like to use my self-hosted gitea server to integrate a comment system to my mkdocs material workflow instead of github.
Could not find any information about gitea support.

Is gitea offially supported?
If so: Where can I find documentation abut it?
If not: Is it at least planned, or even better maybe allready in progress?

Best regards and thank you in davance

By the way:
I also started a discussion on the giscuss project before too.
I am gladly willing to post progression from there into this discussion, if I get any useful advice from there.
--> https://github.com/orgs/giscus/discussions/1580

**Comment by niclasheinz** [comment]
Hi @NejatHakan 

I think you can't use giscus in your Gitea Environment because it uses github's API for discussions. 

But you can integrate Mastodon or similar things for example.

**Reply by kamilkrzyskow** [reply]
Gitea doesn't support discussions there is an open issue for it:
- https://github.com/go-gitea/gitea/issues/17858

The [comments.html](https://github.com/squidfunk/mkdocs-material-insiders/blob/master/src/overrides/partials/comments.html) template is simply a snippet with a HTML embed configured to load the comments integration.

You can look at other alternatives and check how to integrate them;
- https://alternativeto.net/software/giscus/
- https://github.com/utterance/utterances - this one uses Issues, which Gitea does support\*
- https://github.com/isso-comments/isso/ - this is a self-hosted alternative to Disqus, so the comments would be managed outside of Jira-like issues/discussions.

> I also started a discussion on the giscuss project before too.

EDIT: My bad, you've asked on giscus about the gitea support, not on gitea for the giscus support, I somehow mixed it up 😅 
~~Thanks for informing about this, since there is a forum for questions for Gitea users I think you should ask there what the best approach would be to integrate a comment system.~~ This could be a case of XY problem 🤔.

\*There is also the matter of the "Application BOT", which would manage your giscus/utterances actions, and this might be not compatible with Gitea due to expected GitHub APIs like Niclas said.
